src/EnergyPlus/DataSurfaces.cc

```diff
@@ -734,4 +734,34 @@ Real64 AbsBackSide(EnergyPlusData &state, int SurfNum)
     return AbsorptanceFromExteriorBackSide + AbsorptanceFromInteriorBackSide;
 }
 
+void GetVariableAbsorptanceSurfaceList(EnergyPlusData &state)
+{
+    for (int surfNum = 1; surfNum <= state.dataSurface->TotSurfaces; ++surfNum) {
+        auto const thisSurface = state.dataSurface->Surface(surfNum);
+        int ConstrNum = thisSurface.Construction;
+        if (ConstrNum <= 0) continue;
+        auto const &thisConstruct = state.dataConstruction->Construct(ConstrNum);
+        int TotLayers = thisConstruct.TotLayers;
+        if (TotLayers == 0) continue;
+        int materNum = thisConstruct.LayerPoint(1);
+        if (materNum == 0) continue; // error finding material number
+        auto const *thisMaterial = state.dataMaterial->Material(materNum);
+        if (thisMaterial->absorpVarCtrlSignal != Material::VariableAbsCtrlSignal::Invalid) {
+            // check for dynamic coating defined on interior surface
+            if (thisSurface.ExtBoundCond != ExternalEnvironment) {
+                ShowWarningError(state, format("MaterialProperty:VariableAbsorptance defined on an interior surface, {}", thisSurface.Name));
+            } else {
+                state.dataSurface->AllVaryAbsOpaqSurfaceList.push_back(surfNum);
+            }
+        }
+        // check for dynamic coating defined on the non-outside layer of a construction
```

Comment:
This check can be done more efficiently by looping over all constructions (separate from the surface loop) and just throwing an error and mention the construction name.  For both warnings, I would add something like "This will be ignored."

Reply:
right right. Shouldn't nest it in the surface loop.